libsoftwarecontainer/src/gateway/dbus/dbusgatewayinstance.cpp

```diff
@@ -0,0 +1,225 @@
+
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+
+#include "dbusgateway.h"
+#include "dbusgatewayparser.h"
+
+DBusGatewayInstance::DBusGatewayInstance(ProxyType type
+                       , const std::string &gatewayDir
+                       , const std::string &name)
+    : Gateway(ID)
+    , m_type(type)
+{
+    m_socket = gatewayDir
+             + (m_type == SessionProxy ? "/sess_" : "/sys_")
+             + name
+             + ".sock";
+
+    // Write configuration
+    m_entireConfig = json_object();
+    m_busConfig = json_array();
+
+    const char* unusedTypeStr;
+    if (m_type == SessionProxy) {
+        typeStr = SESSION_CONFIG;
+        unusedTypeStr = SYSTEM_CONFIG;
+
+    } else {
+        typeStr = SYSTEM_CONFIG;
+        unusedTypeStr = SESSION_CONFIG;
+    }
+    json_object_set(m_entireConfig, typeStr, m_busConfig);
+
+    // Send an empty array as config for the irrelevant proxy type
+    json_t* emptyArray = json_array();
+    json_object_set(m_entireConfig, unusedTypeStr, emptyArray);
+
+
+}
+
+DBusGatewayInstance::~DBusGatewayInstance()
+{
+    if (isActivated()) {
+        teardown();
+    }
```

Comment:
That is a bad idea since the config will be saved or memory will be leaked when tearing down without deleting the Gateway.